packages/devtools_app/lib/src/screens/debugger/debugger_controller.dart

```diff
@@ -386,7 +379,7 @@ class DebuggerController extends DisposableController
     // serviceManager.isolateManager.selectedIsolateState.isPaused.value;
     // listening for changes there instead of having separate logic.
     await _getStackOperation?.cancel();
-    _isPaused.value = paused;
+    appState.setPaused(paused);
```

Comment:
appState would drive updating whether it is paused itself rather than depending on the debuggerController to drive it.

Reply:
how appState would know about it?

Reply:
From Jacob:

appState would know about it by moving the logic relevant to this into appState instead of debuggerController
there should be a very small number of events that are listened for to determine the paused state for isolates and appState should contain that logic rather than debugController intermediating it

For example, when you get an Isolate object, it tells you the current paused state
and when you get events that you paused at a breakpoint then you are paused

Fwiw, I wouldn't be surprised if debugger_controller has some bugs in how it is tracking the paused state currently as it seems to be doing some slightly unsafe things modifying the paused state based on UI actions rather than notifications from the VM Service

Reply:
Added TODO.